### ISSUE TITLE
HCC V28

### ISSUE THREAD

**Issue by ronnie-canopy** [yubin-park/hccpy#54]
Are there plans to add the V28 updates soon per the advance notice linked below?
https://www.cms.gov/files/document/2024-advance-notice.pdf.

I am happy to contribute to this work if updates are planned.

**Comment by dr00b** [comment]
@ronnie-canopy Yes, we intend to start work as soon as the SAS model is released. In the interim, hoping to crowdsource a bit to get to an exhaustive, high quality test-set. Stay tuned!

**Comment by dr00b** [comment]
@ronnie-canopy  see #56 :)

**Comment by ronnie-canopy** [comment]
I've already added code to the Python for a V28 using the model that was released based on the Advance Notice.
I've tested this on a sample of about 5000 members (no ESRD and no C-SNP) and I match on all but 4 cases against the output of an Excel V28 HCC model from EpiSource.

The 4 cases are where we suspect EpiSource did not code for the footnote on page 53 of the advance notice
I coded for this ‘interaction’ based on CMS SAS macro for the model V2823T2M:
            %*heart interaction patch;
            IF CC223 = 1 AND MAX(CC221, CC222, CC224, CC225, CC226) = 0                                                                                                                                          
            THEN CC223 = 0;                                                                                                                                                                     

![image](https://user-images.githubusercontent.com/84885071/232832764-30399185-1dfd-4503-a96b-1e59c5eae48d.png)

I will try to share in a few days - I'm new to GitHub and not a seasoned Python programmer.
I've also altered the code to build in the normalization factors and the MA coding pattern adjustment, and thus also present a pricing RAF in the JSON output.
